test/dlc_tests/conftest.py

```diff
@@ -643,6 +677,25 @@ def generate_unique_values_for_fixtures(metafunc_obj, images_to_parametrize, val
     return fixtures_parametrized
 
 
+def lookup_condition(lookup, image):
+    """
+    Return true if the ECR repo name ends with the lookup or lookup contains job type or device type part of the image uri.
+    """
+    #Extract ecr repo name from the image and check if it exactly matches the lookup (fixture name)
+    repo_name = image.split("/")[-1].split(":")[0]
+
+    job_type = ("training", "inference",)
+    device_type = ("cpu", "gpu", "eia", "neuron", "hpu")
+
+    if not repo_name.endswith(lookup):
+        if (lookup in job_type or lookup in device_type) and lookup in image:
```

Comment:
I think this condition can be fragile and is too linked to assumptions about what the fixtures can be. Let's say we want to add a lookup called "pytorch" - which runs on all pytorch images. Example would be torchaudio sanity test. We'd need to add fixture and also update this condition.

I'm a little concerned about this change because it has the ability to affect all the testing logic under the hood - and I don't want to get into a situation where we aren't running the tests we expect to be running. Have you verified on all test builds that tests are being run as expected? 

I have less context on things like EIA, which I assume could potentially also rely on `<framework>_<job_type>` fixture.

If we make this change, let's also formalize the assumptions in our README
- example: for new processors outside of cpu/gpu/eia/neuron/hpu, please name your repo as `<framework>-<job_type>-<processor>`

Reply:
For the first part, yes we need to add condition that matched the framework name. 
For the second part, the repo names were already in the format of `<framework>-<job_type>-<processor>`. We just were not using it, the way we had is `<framework>-<job_type>` and a fixture like `<processor>_only` which is unnecessary. This logic fixes  it.